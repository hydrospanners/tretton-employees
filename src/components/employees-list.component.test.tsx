import renderer from "react-test-renderer";
import { EmployeesListComponent } from "./employees-list.component";

describe("GIVEN the EmployeesList component", () => {
  test("WHEN the array is empty THEN the list should display empty", () => {
    const rendered = renderer.create(
      <EmployeesListComponent employees={[]} loading={false} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
  test("WHEN the array has items THEN the list should display employees", () => {
    const rendered = renderer.create(
      <EmployeesListComponent
        employees={[
          {
            name: "John",
            area: "IT",
            email: "some@email.com",
            phoneNumber: "1234567890",
            gitHub: "somegithub",
            imagePortraitUrl: "/someurl.jpg",
            imageWallOfLeetUrl: "someotherurl",
            linkedIn: "someotherlinkedin",
            twitter: "sometwitter",
            mainText: "sometext",
            manager: "somemanager",
            office: "someoffice",
            orgUnit: "someorgunit",
            primaryRole: "someprimaryrole",
            published: true,
            secondaryRole: "somesecondaryrole",
            stackOverflow: "somestackoverflowurl",
            highlighted: false,
          },
        ]}
        loading={false}
      />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
