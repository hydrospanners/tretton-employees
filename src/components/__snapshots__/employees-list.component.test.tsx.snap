// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN the EmployeesList component WHEN the array has items THEN the list should display employees 1`] = `
<ul
  className="MuiList-root MuiList-padding css-1mcnwpj-MuiList-root"
>
  <a
    className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
    href="/employee/some@email.com"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    <div
      className="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
      >
        <img
          alt="John"
          data-nimg="1"
          decoding="async"
          height={75}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fsomeurl.jpg&w=256&q=75"
          srcSet="/_next/image?url=%2Fsomeurl.jpg&w=96&q=75 1x, /_next/image?url=%2Fsomeurl.jpg&w=256&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={75}
        />
      </div>
    </div>
    <div
      className="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        John
      </span>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
      >
        some@email.com
      </p>
    </div>
  </a>
</ul>
`;

exports[`GIVEN the EmployeesList component WHEN the array is empty THEN the list should display empty 1`] = `
<ul
  className="MuiList-root MuiList-padding css-1mcnwpj-MuiList-root"
/>
`;
